### DISCUSSION TITLE
Incrementalxchacha20poly1305

### DISCUSSION THREAD

**Discussion by sidielnikov** [samuel-lucas6/Kryptor#49] [Q&A]
Hi Samuel I want to know Kryptor in subsequent updates to switch to Incrementalxchacha20poly1305?

**Comment by samuel-lucas6** [comment]
Good question! I will have to think about it for the future, but it would be a breaking change. It also depends if I want to allow seeking in the future and turn Kryptor into a library like age, which would certainly improve code quality but take significant time to do.

My university work only gets more intense from here on out, so I can't make any promises about when new versions will be released.

**Reply by sidielnikov** [reply]
If all this happens and the kryptor switches to Incrementalxchachacha20poly1305 will this security or affect the encryption speed, unlike the most difficult Chacha20poly1305.encrypt?

**Reply by samuel-lucas6** [reply]
Not really as it's just calling ChaCha20-Poly1305 internally, although I need to double check if it's key committing.
